_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -76,35 +89,73 @@ Label that is coded with the `p` element and associated by the aria-labelledby a
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
-#### Passed example 3
+#### Passed Example 3
 
 Implicit label that is coded with the `label` element and describes the purpose of the associated element.
 
 ```html
 <label>First name:<input id="fname" type="text" name="fname"/></label>
 ```
 
-#### Passed example 4
+#### Passed Example 4
 
-Label is visible, but not included in accessibility tree
+Label is visible, but not included in accessibility tree.
 
 ```html
 <p id="label_fname" aria-hidden="true">First name:</p>
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
-#### Passed example 5
+#### Passed Example 5
 
-Label is included in accessibility tree, but not visible
+Label is included in accessibility tree, but not visible.
 
 ```html
 <p id="label_fname" style="position: absolute; top: -9999px; left: -9999px;">First name:</p>
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
+#### Passed Example 6
+
+Programatically determinable headings provide a context that differentiates the purpose of the otherwise identically named form fields.
+
+```html
+<h2>Shipping address</h2>
+<label>Name<input id="name" type="text" name="name"/></label>
+<label>Street<input id="street" type="text" name="street"/></label>
+
+<h2>Billing address</h2>
+<label>Name<input id="name" type="text" name="name"/></label>
+<label>Street<input id="street" type="text" name="street"/></label>
+````
+
+#### Passed Example 7
+
+Accessible name created through "aria-label" describes the purpose of the associated element.
+
+```html
+<input aria-label="First name" id="fname" type="text" name="fname"/>
+```
+
+#### Passed Example 8
+
+Otherwise identically named form fields can be differentiated both visually and in a way that is included in the accessibility tree, though different methods are used for each purpose.
+
+```html
+<h2 aria-hidden="true">Shipping address</h2>
+<p id="shipping" style="position: absolute; top: -9999px; left: -9999px;">(Shipping)</p>
+<label>Name<input id="name" type="text" name="name" aria-describedby="shipping" /></label>(Shipping
```

Comment:
I believe you don't want the `(Shipping` here

Reply:
Will it work if we do this instead? 

`<p id="shipping" style="display:none">(Shipping)</p>`

I *think* even elements that are excluded from the accessibility tree will still be used for the accessible description computation if references with aria-describedby.

@kasperisager, I did look in https://www.w3.org/TR/accname-1.1/, but am hoping you can confirm my interpretation :)

Reply:
Can confirm:

> If the current node is hidden and is not directly referenced by `aria-labelledby` or `aria-describedby,` nor directly referenced by a native host language text alternative element (e.g. `label` in HTML) or attribute, return the empty string.

Reply:
:-) 
My issue was not the Shipping in this line `<p id="shipping" style="display:none">(Shipping)</p>`
Your change works but I'm okay with how it was before
My issue was with the (Shipping in the following line `<label>Name<input id="name" type="text" name="name" aria-describedby="shipping" /></label>(Shipping` where it shows up outside the `label` element

Reply:
Aha! Looks like a rogue copy/paster leftover. I have removed it now, since it doesn't make sense. But kept the rest, since I am not quite sure how it was before...